code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm

```diff
@@ -397,6 +397,10 @@
 
 			if(!primary_weapon && isgun(thing))
 				var/obj/item/weapon/gun/thing_gun = thing
+				for(var/item in to_pickup)
```

Comment:
i assume picking up multiple guns (trying to anyway) bugged the AI?

Reply:
Yeah, it did, thought this way is the best option to make them not.